test/Chainweb/Test/Pact/PactMultiChainTest.hs

```diff
@@ -822,6 +825,73 @@ pact46UpgradeTest = do
         [ "(pairing-check [{'x: 1, 'y: 2}] [{'x:[0, 0], 'y:[0, 0]}])"
         ])
 
+chainweb219UpgradeTest :: PactTestM ()
+chainweb219UpgradeTest = do
+
+  -- run past genesis, upgrades
+  runToHeight 69
+
+  -- Block 70, pre-fork, no errors on chain
+  runBlockTest
+      [ PactTxTest addErrTx $
+        assertTxFailure
+        "Should fail on + with incorrect argument types"
+        ""
+      , PactTxTest mapErrTx $
+        assertTxFailure
+        "Should fail on map with the second argument not being a list"
+        ""
+      , PactTxTest nativeDetailsTx $
+        assertTxSuccessWith
+        "Should print native details on chain"
+        (pString nativeDetailsMsg)
+        (over (_PLiteral . _LString) (T.take (T.length nativeDetailsMsg)))
+      , PactTxTest fnDetailsTx $
+        assertTxSuccessWith
+        "Should display function preview string"
+        (pString fnDetailsMsg)
+        (over (_PLiteral . _LString) (T.take (T.length fnDetailsMsg)))
+      ]
+
+  -- Block 71, post-fork, errors should return on-chain but different
+  runBlockTest
+      [ PactTxTest addErrTx $
+        assertTxFailureWith
+        "Should error with the argument types in +"
+        (isPrefixOf "Invalid arguments in call to +, received arguments of type" . show)
+      , PactTxTest mapErrTx $
+        assertTxFailure
+        "Should fail on map with the second argument not being a list"
+        "map: expecting list, received argument of type: string"
+      , PactTxTest nativeDetailsTx $
+        assertTxFailure
+        "Should print native details on chain"
+        "Cannot display native function details in non-repl context"
+      , PactTxTest fnDetailsTx $
+        assertTxFailure
+        "Should not display function preview string"
+        "Cannot display function details in non-repl context"
+      ]
+  where
+    addErrTx = buildBasicGas 10000
+        $ mkExec' (mconcat
+        [ "(+ 1 \"a\")"
+        ])
+    mapErrTx = buildBasicGas 10000
+        $ mkExec' (mconcat
+        [ "(map (+ 1) \"a\")"
+        ])
+    nativeDetailsMsg = "native `=`  Compare alike terms for equality"
+    nativeDetailsTx = buildBasicGas 10000
+        $ mkExec' (mconcat
+        [ "="
+        ])
+    fnDetailsMsg = "(defun coin.transfer:string"
```

Comment:
for anyone else reviewing, this is not a typo it's just a prefix

Reply:
:+1: Yeah. This is a prefix because the actual string is pretty long. Same with the native docs.